src/ripple/rpc/handlers/Submit.cpp

```diff
@@ -142,12 +142,41 @@ Json::Value doSubmit (RPC::Context& context)
         {
             std::string sToken;
             std::string sHuman;
+            std::uint32_t mAccountSeqNext;
+            std::uint32_t mAccountSeqAvail;
+            bool mAccepted;
+            bool mApplied;
+            bool mBroadcast;
+            bool mQueued;
+            bool mKept;
 
             transResultInfo (tpTrans->getResult (), sToken, sHuman);
 
             jvResult[jss::engine_result]           = sToken;
             jvResult[jss::engine_result_code]      = tpTrans->getResult ();
             jvResult[jss::engine_result_message]   = sHuman;
+
+            mAccepted = tpTrans->getAccepted (mApplied, mBroadcast,
+                    mQueued, mKept);
```

Comment:
I'd get rid of the combined return value/out parameters for `getAccepted` and just return a struct with the values you want. You can add a function to the struct to return true if any of the values are set. Something like:
```
            auto const accepted = tpTrans->getAccepted();

            jvResult[jss::accepted] = accepted.any();
            jvResult[jss::applied] = accepted.applied;
            jvResult[jss::broadcast] = accepted.broadcast;
            jvResult[jss::queued] = accepted.queued;
            jvResult[jss::kept] = accepted.kept;
```

Reply:
Done.